tst/EnergyPlus/unit/RoomAirflowNetwork.unit.cc

```diff
@@ -579,3 +600,142 @@ TEST_F(EnergyPlusFixture, RoomAirInternalGains_InternalHeatGains_Check)
+    if (check) {
+        EXPECT_EQ("SupplyNode", SupplyNodeName);
+        EXPECT_EQ("ReturnNode", ReturnNodeName);
+    }
+
+    zoneEquipType = DataZoneEquipment::ZoneEquipType::OutdoorAirUnit;
+    state->dataOutdoorAirUnit->OutAirUnit.allocate(1);
+    state->dataOutdoorAirUnit->OutAirUnit(EquipIndex).AirOutletNode = 1;
+    state->dataOutdoorAirUnit->OutAirUnit(EquipIndex).AirInletNode = 2;
+    check = CheckEquipName(*state, EquipName, SupplyNodeName, ReturnNodeName, EquipIndex, zoneEquipType);
+    if (check) {
+        EXPECT_EQ("SupplyNode", SupplyNodeName);
+        EXPECT_EQ("ReturnNode", ReturnNodeName);
+    }
+
+    zoneEquipType = DataZoneEquipment::ZoneEquipType::PackagedTerminalAirConditioner;
+    UnitarySystems::UnitarySys thisUnit;
+    state->dataUnitarySystems->unitarySys.push_back(thisUnit);
+    state->dataUnitarySystems->getInputOnceFlag = false;
+    state->dataUnitarySystems->unitarySys[EquipIndex - 1].Name = EquipName;
+    state->dataUnitarySystems->numUnitarySystems = 1;
+    state->dataUnitarySystems->unitarySys[EquipIndex - 1].AirOutNode = 1;
+    state->dataUnitarySystems->unitarySys[EquipIndex - 1].AirInNode = 2;
+    check = CheckEquipName(*state, EquipName, SupplyNodeName, ReturnNodeName, EquipIndex, zoneEquipType);
+    if (check) {
+        EXPECT_EQ("SupplyNode", SupplyNodeName);
+        EXPECT_EQ("ReturnNode", ReturnNodeName);
+    }
+
+    zoneEquipType = DataZoneEquipment::ZoneEquipType::PackagedTerminalHeatPump;
+    // UnitarySystems::UnitarySys thisUnit;
+    // state->dataUnitarySystems->unitarySys.push_back(thisUnit);
+    state->dataUnitarySystems->getInputOnceFlag = false;
+    state->dataUnitarySystems->unitarySys[EquipIndex - 1].Name = EquipName;
+    state->dataUnitarySystems->numUnitarySystems = 1;
+    state->dataUnitarySystems->unitarySys[EquipIndex - 1].AirOutNode = 1;
+    state->dataUnitarySystems->unitarySys[EquipIndex - 1].AirInNode = 2;
+    check = CheckEquipName(*state, EquipName, SupplyNodeName, ReturnNodeName, EquipIndex, zoneEquipType);
+    if (check) {
+        EXPECT_EQ("SupplyNode", SupplyNodeName);
+        EXPECT_EQ("ReturnNode", ReturnNodeName);
+    }
+    zoneEquipType = DataZoneEquipment::ZoneEquipType::PackagedTerminalHeatPumpWaterToAir;
+    // UnitarySystems::UnitarySys thisUnit;
+    // state->dataUnitarySystems->unitarySys.push_back(thisUnit);
+    state->dataUnitarySystems->getInputOnceFlag = false;
+    state->dataUnitarySystems->unitarySys[EquipIndex - 1].Name = EquipName;
+    state->dataUnitarySystems->numUnitarySystems = 1;
+    state->dataUnitarySystems->unitarySys[EquipIndex - 1].AirOutNode = 1;
+    state->dataUnitarySystems->unitarySys[EquipIndex - 1].AirInNode = 2;
+    check = CheckEquipName(*state, EquipName, SupplyNodeName, ReturnNodeName, EquipIndex, zoneEquipType);
+    if (check) {
+        EXPECT_EQ("SupplyNode", SupplyNodeName);
+        EXPECT_EQ("ReturnNode", ReturnNodeName);
+    }
```

Comment:
These last 2 unit tests are looking at the PTAC unitarySys object (report the object type and name to find out) and since you are using the same node names and node 1 and 2, all these results give the same answer. This unit test should use unique node names and multiple PTUnits and UnitarySystems. For zone equipment, PTAC, PTHP, PTWSHP and UnitarySystem all have indexes 1 - n.

Reply:
@rraustad Although I use the same name, I did debug it and found they are correct.